web-local/src/main/java/com/bazaarvoice/emodb/local/EmoServiceWithZK.java

```diff
@@ -152,4 +202,84 @@ private static boolean isLocalZooKeeperRunning() {
+            @Override
+            protected void configure() {
+                bind(String.class).annotatedWith(ServerCluster.class).toInstance(cluster);
+                bind(ServerFactory.class).toInstance(emoConfig.getServerFactory());
+                bind(ApiKeyEncryption.class).asEagerSingleton();
+                install(new SelfHostAndPortModule());
+            }
+        };
+
+        Injector injector = Guice.createInjector(module);
+        HostAndPort selfHostAndPort = injector.getInstance(Key.get(HostAndPort.class, SelfHostAndPort.class));
+        ApiKeyEncryption apiKeyEncryption = injector.getInstance(ApiKeyEncryption.class);
+        String adminApiKey = emoConfig.getAuthorizationConfiguration().getAdminApiKey();
+        try {
+            adminApiKey = apiKeyEncryption.decrypt(adminApiKey);
+        } catch (Exception e) {
+            if (ApiKeyEncryption.isPotentiallyEncryptedApiKey(adminApiKey)) {
+                throw e;
+            }
+        }
+
+        // Create a client for the local EmoDB service
+        UserAccessControl uac = ServicePoolBuilder.create(UserAccessControl.class)
+                .withHostDiscoverySource(new UserAccessControlFixedHostDiscoverySource("http://localhost:" + selfHostAndPort.getPort()))
+                .withServiceFactory(UserAccessControlClientFactory.forCluster(cluster, metricRegistry).usingCredentials(adminApiKey))
+                .withMetricRegistry(metricRegistry)
+                .buildProxy(new ExponentialBackoffRetry(5, 50, 1000, TimeUnit.MILLISECONDS));
+
+        try {
+            for (String permissionsYaml : permissionsYamls) {
+                Map<String, List<String>> permissions;
+                try {
+                    permissions = objectMapper.readValue(new File(permissionsYaml), new TypeReference<Map<String, List<String>>>() {});
+                } catch (Exception e) {
+                    System.err.println("Failed to load permissions from file " + permissionsYaml);
+                    e.printStackTrace(System.err);
+                    return;
+                }
+
+                // Use the client to create or update all roles with permissions from the file
+                for (Map.Entry<String, List<String>> entry : permissions.entrySet()) {
+                    RoleIdentifier roleIdentifier = RoleIdentifier.fromString(entry.getKey());
+                    EmoRoleKey roleKey = new EmoRoleKey(roleIdentifier.getGroup(), roleIdentifier.getId());
+                    if (uac.getRole(roleKey) == null) {
+                        uac.createRole(new CreateEmoRoleRequest(roleKey)
+                                .setPermissions(ImmutableSet.copyOf(entry.getValue())));
+                    } else {
+                        uac.updateRole(new UpdateEmoRoleRequest(roleKey)
+                                .grantPermissions(ImmutableSet.copyOf(entry.getValue())));
+                    }
+                }
+
+            }
```

Comment:
Just a comment - like we create anonymous role which is configured in web-local, why not create api keys tied with these roles. Not sure where or how they can be used though.